rotkehlchen/exchanges/gemini.py

```diff
@@ -554,9 +594,79 @@ def query_online_deposits_withdrawals(
                 continue
 
             movements.append(movement)
-
+            with self.db.user_write() as write_cursor:
+                if len(history_events) != 0:
+                    db = DBHistoryEvents(self.db)
+                    db.add_history_events(write_cursor=write_cursor, history=history_events)
+                self.db.set_dynamic_cache(
+                    write_cursor=write_cursor,
+                    name=DBCacheDynamic.LAST_QUERY_TS,
+                    value=ts_now(),
+                    location=self.location.serialize(),
+                    location_name=self.name,
+                )
```

Comment:
Though the function needs to change a bit more, this is not needed.

1. Stop earlier. Once you know nothing is there, no need to open a DB cursor.
2. The cache is not needed here. In coinbase, it was to remember the last transfer id of the successfully pulled transaction/transfer. That's because the api queries from last transfer. Here that's not the case and the api remembers in a generic way start/end ts for asset movements. These are the arguments to this function.
```suggestion
            if len(history_events) == 0:
                return movements

            with self.db.user_write() as write_cursor:
                    db = DBHistoryEvents(self.db)
                    db.add_history_events(write_cursor=write_cursor, history=history_events)
```